packages/next/src/server/dynamic-rendering-utils.ts

```diff
@@ -63,7 +66,21 @@ export function makeHangingPromise<T>(
     // we attach a noop catch handler here to suppress this warning. If you actually await somewhere or construct
     // your own promise out of it you'll need to ensure you handle the error when it rejects.
     hangingPromise.catch(ignoreReject)
-    return hangingPromise
+
+    return new Proxy(hangingPromise, {
+      ...handler,
+      get: function get(target, prop, receiver) {
+        if (prop === 'then' || prop === 'catch' || prop === 'finally') {
+          // TODO: Maybe annotate dynamic access here.
+        }
+
+        if (handler?.get) {
+          return handler.get(target, prop, receiver)
+        }
+
+        return ReflectAdapter.get(target, prop, receiver)
+      },
+    })
```

Comment:
I'd just implement this in params and leave the other ones alone. This will be too tempting in the future. It's hard to get proxies just right and they aren't fast. So if we can get away with normal property overriding then we should try